### DISCUSSION TITLE
pg_net extension availability within docker image

### DISCUSSION THREAD

**Discussion by bhvngt** [supabase/supabase#3435] [Questions] [answered]
Do you have plans to add `pg_net` extension as part of supabase postgres docker image? It does not seem to be available based on the extensions listed on the [supabase/postgres](https://hub.docker.com/r/supabase/postgres) mage page.

**Comment by soedirgo** [comment, marked as answer]
This is now available starting from supabase/postgres:14.1.0

**Comment by bhvngt** [comment]
Thanks @soedirgo.

I have pulled the postgres:14.1.0 image and ran following commands

```sql
CREATE EXTENSION pg_net; -- success
select
    net.http_post(
            url:='https://httpbin.org/post',
            body:='{"hello": "world"}'::jsonb
        ) as request_id; 
-- returns request_id 1
select
    (response).body::json
from
    net.http_collect_response(request_id:=1);
-- returns body as <null>
```

I get empty response. When I query `net._http_response` table, I see `Timeout was reached` error message. I tried running following command inside `postgres` container `curl -X POST -d '{"hello": "world"}' https://httpbin.org/post`. Its working as expected.

Just wanted to check if I am doing something wrong here?

**Reply by soedirgo** [reply]
Can you try setting the timeout?
```sql
select
    net.http_post(
            url:='https://httpbin.org/post',
            body:='{"hello": "world"}'::jsonb,
            timeout_milliseconds:=0 -- no timeout
        ) as request_id;
```
Odd that it says `SUCCESS` though - that should be an error.

**Reply by bhvngt** [reply]
This actually works. Without timeout, it gives me blank result. 

Is this how we are supposed to post requests?

**Reply by soedirgo** [reply]
The default timeout is 1 second. Due to how it's implemented, it's not performed immediately, so that might be why it's not enough.

**Reply by bhvngt** [reply]
ohh. ok.. Looks like I need to increase the default timeout. Like you said, earlier, I would have expected status as `ERROR`.
